cli/src/pcluster/config/config_patch.py

```diff
@@ -137,15 +137,36 @@ def _compare_section(self, base_section: dict, target_section: dict, section_sch
                             )
             else:
                 # Simple param
-                target_value = target_section.get(data_key, None) if target_section else None
-                base_value = base_section.get(data_key, None) if base_section else None
+                target_value, target_data_key = self._get_value_from_section(data_key, target_section)
+                base_value, base_data_key = self._get_value_from_section(data_key, base_section)
+                if target_data_key != base_data_key:
+                    # So far, this only happens when custom actions scripts are changed across simple and sequence.
+                    data_key = param_path.pop()
+                    base_value = {base_data_key: base_value}
+                    target_value = {target_data_key: target_value}
+                else:
+                    # So far, target_data_key is different from data_key only when
+                    # custom actions scripts are in sequence.
+                    data_key = target_data_key
```

Comment:
This change is for the clearness of the change output. It does little to improve correctness